tools/tf_env_collect.sh

```diff
@@ -52,84 +67,86 @@ platform.python_compiler(),
 platform.python_implementation(),
 ))
 EOF
-${python_bin_path} /tmp/check_python.py 2>&1  >> ${OUTPUT_FILE}
 
-{
-echo
-echo '== check os platform ==============================================='
-} >> ${OUTPUT_FILE}
+  print_header "check os platform"
 
-cat <<EOF > /tmp/check_os.py
+  "${PYTHON_BIN_PATH}" <<EOF
 import platform
 
-print("""os: %s
-os kernel version: %s
-os release version: %s
-os platform: %s
-linux distribution: %s
-linux os distribution: %s
-mac version: %s
-uname: %s
-architecture: %s
-machine: %s
-""" % (
-platform.system(),
-platform.version(),
-platform.release(),
-platform.platform(),
-platform.linux_distribution(),
-platform.dist(),
-platform.mac_ver(),
-platform.uname(),
-platform.architecture(),
-platform.machine(),
-))
+PLATFORM_ENTRIES = [
+    ("os", "system"),
+    ("os kernel version", "version"),
+    ("os release version", "release"),
+    ("os platform", "platform"),
+    ("freedesktop os release", "freedesktop_os_release"),
+    ("mac version", "mac_ver"),
+    ("uname", "uname"),
+    ("architecture", "architecture"),
+    ("machine", "machine"),
+]
+
+for label, function_name in PLATFORM_ENTRIES:
+    if hasattr(platform, function_name):
+        function = getattr(platform, function_name)
+        result = function()  # Call the function
+        print(f"{label}: {result}")
+    else:
+        print(f"{label}: N/A")
 EOF
-${python_bin_path} /tmp/check_os.py 2>&1  >> ${OUTPUT_FILE}
 
-{
-  echo
-  echo '== are we in docker ============================================='
-  num=`cat /proc/1/cgroup | grep docker | wc -l`;
-  if [ $num -ge 1 ]; then
+
+  print_header 'are we in docker'
+  if grep -q docker /proc/1/cgroup; then
     echo "Yes"
   else
     echo "No"
   fi
   
-  echo
-  echo '== compiler ====================================================='
-  c++ --version 2>&1
-  
-  echo
-  echo '== check pips ==================================================='
-  pip list 2>&1 | grep "proto\|numpy\|tensorflow"
+  print_header 'c++ compiler'
+  if which c++; then
+    c++ --version 2>&1
+  else
+    echo "Not found"
+  fi
   
+  print_header 'check pips'
+  pip list 2>&1 | grep -E 'proto|numpy|tensorflow|tf_nightly'
   
-  echo
-  echo '== check for virtualenv ========================================='
-  ${python_bin_path} -c "import sys;print(hasattr(sys, \"real_prefix\"))"
```

Comment:
This hasn't been a valid way to check if you're in a venv since the invention of python 3.